portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,34 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  public static final Gson GSON = new Gson();
```

Comment:
why public?

Reply:
I'm thinking that if I don't want to initialize gson every time I want to use it in my whole project, I should make it accessible to all classes, so I put a public there.
Will there be difference if I do `public static final` or `static final`?

Reply:
actually, I means ```private static final Gson ...```.

For your question:
1. Using ```static``` basically solve your first question. static variable won't be initialized every time you use.
2. In most cases, I don't think having duplicate static variables is a critical problem. If you really want to ensure that only one instance in the whole project, I would suggest create a package with a class to store all common variables. There you can make all common variables public.
Otherwise, accessing static variable in a irrelevant class is confusing.
i.e
```
class DataServlet {
  public static final Gson GSON....
}
class Storage {
  public void read() {
    DataServlet.GSON ...
  }
}
```
3. IIRC, by default, visibility of java class member variables is between public & private. It's only visible to classes within the same package.

Reply:
Sure.